services/ui-src/src/components/export/ExportedEntityDetailsOverlaySection.tsx

```diff
@@ -0,0 +1,275 @@
+import { Fragment, useContext } from "react";
+// components
+import { ExportedSectionHeading, ReportContext } from "components";
+// types, utils
+import {
+  EntityShape,
+  FormField,
+  FormLayoutElement,
+  ModalOverlayReportPageShape,
+  ReportShape,
+  ReportType,
+} from "types";
+// verbiage
+import mcparVerbiage from "../../verbiage/pages/mcpar/mcpar-export";
+import mlrVerbiage from "../../verbiage/pages/mlr/mlr-export";
+import { Box, Heading, Text } from "@chakra-ui/react";
+import { assertExhaustive } from "utils/other/typing";
+import { ExportedEntityDetailsTable } from "./ExportedEntityDetailsTable";
+import uuid from "react-uuid";
+
+const exportVerbiageMap: { [key in ReportType]: any } = {
+  MCPAR: mcparVerbiage,
+  MLR: mlrVerbiage,
+  NAAAR: undefined,
+};
+
+export const ExportedEntityDetailsOverlaySection = ({
+  section,
+  ...props
+}: ExportedEntityDetailsOverlaySectionProps) => {
+  const { report } = useContext(ReportContext);
+  const entityType = section.entityType;
+
+  return (
+    <Box sx={sx.sectionHeading} {...props}>
+      <ExportedSectionHeading
+        heading={exportVerbiageMap[report?.reportType as ReportType]}
+        verbiage={{
+          ...section.verbiage,
+          intro: {
+            ...section.verbiage.intro,
+            info: undefined,
+            exportSectionHeader: undefined,
+          },
+        }}
+      />
+      {renderEntityDetailTables(
+        report?.reportType as ReportType,
+        report?.fieldData[entityType],
+        section,
+        report
+      )}
+    </Box>
+  );
+};
+
+export interface ExportedEntityDetailsOverlaySectionProps {
+  section: ModalOverlayReportPageShape;
+}
+/**
+ * Split a list of form fields by the "sectionHeader" layout element type.
+ * This allows returning distinct tables for each section, rather than one large one.
+ *
+ * In the returned FormField sections, the first element is the section header.
+ *
+ * @param formFields List of form fields, possibly containing section headers
+ * @returns array of arrays containing form field elements representing a section
+ */
+export function getFormSections(
+  formFields: (FormField | FormLayoutElement)[]
+): (FormLayoutElement | FormField)[][] {
+  const formSections: FormField[][] = [];
+  const sectionHeaderIndexes: number[] = [];
+
+  for (const [index, field] of formFields.entries()) {
+    if (field.type === "sectionHeader") {
+      sectionHeaderIndexes.push(index);
+    }
+  }
+
+  // Split each section of the form into its own table
+  for (let i = 0; i < sectionHeaderIndexes.length; i++) {
+    const left = sectionHeaderIndexes[i];
+    const right =
+      i + 1 < sectionHeaderIndexes.length
+        ? sectionHeaderIndexes[i + 1]
+        : undefined;
+    formSections.push(formFields.slice(left, right) as FormField[]);
+  }
+
+  if (formSections.length > 0) {
+    return formSections;
+  } else {
+    return [formFields];
+  }
```

Comment:
I _think_ this code would do the same thing without the index manipulation? And it's easier for me to follow, although I don't know if you'll agree.

```tsx
  const formSections: (FormLayoutElement | FormField)[][] = [];

  let currentSection: (FormField|FormLayoutElement)[] = [];
  for (let field of formFields) {
    if (field.type === "sectionHeader") {
      if (currentSection.length > 0) {
        formSections.push(currentSection);
      }
      currentSection = [field];
    }
    else {
      currentSection.push(field);
    }
  }
  formSections.push(currentSection);

  return formSections;
  ```

Reply:
By the way, here are the tests I used to prove this out:

```ts
describe("getFormSections", () => {
  test("it does not error on empty arrays", () => {
    const sections = getFormSections([]);
    expect(sections).toHaveLength(1);
    expect(sections[0]).toHaveLength(0);
  });

  test("it does not split if there is no splitting to do", () => {
    const sections = getFormSections([
      {id: "a", type: "sectionHeader"},
      {id: "b", type: "checkbox"},
    ]);
    expect(sections).toHaveLength(1);
    expect(sections[0].map(({id}) => id)).toEqual(["a", "b"]);
  });

  test("it splits out a new subsection for each header", () => {
    const sections = getFormSections([
      {id: "a", type: "sectionHeader"},
      {id: "b", type: "checkbox"},
      {id: "c", type: "checkbox"},
      {id: "d", type: "sectionHeader"},
      {id: "e", type: "sectionHeader"},
      {id: "f", type: "checkbox"},
    ]);
    expect(sections).toHaveLength(3);
    expect(sections[0].map(({id}) => id)).toEqual(["a", "b", "c"]);
    expect(sections[1].map(({id}) => id)).toEqual(["d"]);
    expect(sections[2].map(({id}) => id)).toEqual(["e", "f"]);
  });
});
```